apps/admin-x-settings/src/components/settings/email/newsletters/NewsletterDetailModal.tsx

```diff
@@ -539,28 +532,39 @@ const NewsletterDetailModalContent: React.FC<{newsletter: Newsletter; onlyOne: b
             const previousFrom = renderSenderEmail(updatedNewsletter, config, defaultEmailAddress);
             const previousReplyTo = renderReplyToEmail(updatedNewsletter, config, supportEmailAddress, defaultEmailAddress) || previousFrom;
 
+            /* TODO: Remove title, prompt, NiceModal */
             let title;
             let prompt;
+            let toastMessage;
 
             if (emailToVerify && emailToVerify === 'sender_email') {
                 title = 'Confirm newsletter email address';
                 prompt = <>We&lsquo;ve sent a confirmation email to <strong>{formState.sender_email}</strong>. Until the address has been verified, newsletters will be sent from the {updatedNewsletter.sender_email ? ' previous' : ' default'} email address{previousFrom ? ` (${previousFrom})` : ''}.</>;
+                toastMessage = <div>We&lsquo;ve sent a confirmation email to <strong>{formState.sender_email}</strong>.</div>;
             } else if (emailToVerify && emailToVerify === 'sender_reply_to') {
                 title = 'Confirm reply-to address';
                 prompt = <>We&lsquo;ve sent a confirmation email to <strong>{formState.sender_reply_to}</strong>. Until the address has been verified, replies will continue to go to {previousReplyTo}.</>;
+                toastMessage = <div>We&lsquo;ve sent a confirmation email to <strong>{formState.sender_reply_to}</strong>.</div>;
             }
 
             if (title && prompt) {
-                NiceModal.show(ConfirmationModal, {
-                    title: title,
-                    prompt: prompt,
-                    cancelLabel: '',
-                    onOk: (confirmModal) => {
-                        confirmModal?.remove();
-                        modal.remove();
-                        updateRoute('newsletters');
-                    }
-                });
+                if (newEmailAddressesFlag) {
```

Comment:
```suggestion
                if (newEmailAddressesFlag || isManagedEmail(config)) {
```

This flag for now won't be enabled on release and is made for self hosters. For Pro, we need to look at the config to see if the email addresses are managed (= using the new DMARC alignment setup) or not.